FeedService/Services/PiletService.cs

```diff
@@ -0,0 +1,130 @@
+      var piletResponse = new PiletApiResponse();
+
+      foreach (var pilet in latestPiletsInfo)
+      {
+        var version = ExtractVersion(pilet);
+        var name = ExtractName(pilet);
+        var url = "http://" + Environment.GetEnvironmentVariable("WEBSITE_HOSTNAME") + "/api/pilet/" + name + "/" + version + "/index.js";
+
+        piletResponse.Items.Add(new PiletMetaData
+        {
+          Name = name,
+          Link = url,
+          Version = version
+
+        });
+      }
+      return piletResponse;
+    }
+
+    public async Task<Stream> GetPiletFile(string fileName, string version, string piletName)
+    {
+      return await _storageAccountRepository.GetFile(fileName, version, piletName);
+    }
+
+    public Task<bool> PublishPilet(byte[] fileData)
+    {
+      try
+      {
+        var files = ExtractFiles(fileData);
+        _storageAccountRepository.UploadFiles(files);
+      }
+      catch (Exception)
+      {
+        return Task.FromResult(false);
+      }
+
+      return Task.FromResult(false);
+    }
+
+    public List<PackageFile> ExtractFiles(byte[] fileData)
+    {
+      var packageJsonFile = new PackageJson();
+      var files = new List<PackageFile>();
+      Stream inStream = new MemoryStream(fileData);
+      Stream gzipStream = new GZipInputStream(inStream);
+      using (var tarInputStream = new TarInputStream(gzipStream, null))
+      {
+        TarEntry entry;
+        while ((entry = tarInputStream.GetNextEntry()) != null)
+        {
+          using (var fileContents = new MemoryStream())
+          {
+            tarInputStream.CopyEntryContents(fileContents);
+            var fileName = GetFileName(entry.Name);
+            if (fileName == "package.json")
+            {
+              packageJsonFile = ParsePackageJson(fileContents, entry.Name);
+            }
+
+            files.Add(new PackageFile
+            {
+              FileName = entry.Name,
+              ContentArray = fileContents.GetBuffer()
+            });
+          }
+        }
+      }
+
+      SetFilePath(files, packageJsonFile);
+      return files;
+    }
+
+    public PackageJson ParsePackageJson(MemoryStream fileContents, string fileFullName)
+    {
+      var content = Encoding.UTF8.GetString(fileContents.GetBuffer());
+      return JsonSerializer.Deserialize<PackageJson>(content);
+    }
+
+
+    private void SetFilePath(List<PackageFile> files, PackageJson packageJsonFile)
+    {
+      foreach (var file in files)
+      {
+        file.Path = packageJsonFile.Name + "/" + packageJsonFile.Version + "/" + file.FileName;
+      }
+    }
+
+    private string GetFileName(string path)
+    {
+      var s = path.Split('/');
+      return s[s.Length - 1];
+    }
+
+    private string ExtractName(string pilet)
+    {
+      return pilet.Split('/')[0];
+    }
+
+    private string ExtractVersion(string pilet)
+    {
+      return pilet.Split('/')[1].Substring(0, 5);
```

Comment:
How is the `0, 5` derived here? Is this due to `0.1.2`? But what if the version is `0.0.1001` or `12.19.201`?